### ISSUE TITLE
CSS, JS, and HTML sniffs

### ISSUE THREAD

**Issue by eugenegorelikov** [WordPress/WordPress-Coding-Standards#1814]
Wordpress themes and plugins are not about just the PHP.

Is is possible to add sniff for HTML, CSS and JS as well?

**Comment by NielsdeBlaauw** [comment]
Hi @eugenegorelikov 

For JavaScript coding standards I can point you to [the WordPress ESLint package](https://www.npmjs.com/package/@wordpress/eslint-plugin).

For CSS coding standards there is an [Stylelint configuration](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress) available.

For HTML I'm not aware of any static analysis tools that can help you. After rendering the page you can obviously use the W3C Validator tool, lighthouse or similar to make some useful analysis of the end result.

**Comment by NielsdeBlaauw** [comment]
After some searching I've found [HTMLHint](https://github.com/htmlhint/HTMLHint) which has some rules applicable to WordPress. And it seems to be able to analyse some dynamic content as well. My test case was suing partial twig files, and it gives some useful pointers.

<img width="1082" alt="Screenshot 2019-10-19 at 20 27 05" src="https://user-images.githubusercontent.com/661876/67149717-e4e5af80-f2ae-11e9-9be1-d0e38efdfd24.png">

I used the following `.htmlhintrc` for my test. These rules catch some of the [WordPress HTML coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/html/) and some other generally incorrect stuff.

```json
{
  "tagname-lowercase": true,
  "attr-lowercase": true,
  "attr-value-double-quotes": true,
  "id-unique": true,
  "src-not-empty": true,
  "attr-no-duplication": true,
  "tag-self-close": true,
  "attr-value-not-empty": true
}
```
